### ISSUE TITLE
Feature: attachment-level storage visibility configuration

### ISSUE THREAD

**Issue by Yoeriwalstra** [czim/laravel-paperclip#57]
Hi @czim,

tl;dr: I want to add an attachment-level configuration option to specify the visibility level of the stored file. This is currently not possible, correct? Any tips for the implementation? You don't happen to be working on this already?

The company I work for uses `codesleeve/laravel-stapler` for working with file attachments in multiple applications. Since this package is no longer maintained, as opposed to yours, we wish to start using `czim/laravel-paperclip`. However, our application requires some files to be publicly accessible (e.g. company content images), but others not (e.g. a user's personal information / resume). 
Unless I missed something in the documentation, it seems to me that the only place where the visibility level of files can be specified currently is Laravel's `config/filesystems.disks` array. This won't do for us, since we need to be able to specify the visibility level per attachment, instead of globally.
Therefore, I suggest adding a `visibility => 'public' || 'private'` key to the `$options` argument of the `PaperclipTrait::hasAttachedFile` method. This option then has to find it's way all the way down to `Czim\FileHandling\Storage\Laravel\LaravelStorage::store` where it can be added as a third argument to `$this->filesystem->put($path, $file->content()[, $visibility])`.
Since I believe this new feature would involve making changes to two different packages (`czim/laravel-paperclip` and `czim/file-handling`), that support a variety of attachment configuration options (`default` object-oriented configuration and legacy `stapler` array configuration), it would be much appreciated if you could give me some pointers for the minimum requirements to make this work, pitfalls and restrictions or, even better, collaborate on it with me.

I hope to hear from you soon.

Kind regards,

yoeri

**Comment by Yoeriwalstra** [comment]
Update: after digging around in the code, I realized I would have to invest more energy in this than I'm willing and able to. Sorry for the mix up.
